lib/transaction/payload/proupregtxpayload.js

```diff
@@ -6,57 +6,58 @@ var Preconditions = require('../../util/preconditions');
 var BufferWriter = require('../../encoding/bufferwriter');
 var BufferReader = require('../../encoding/bufferreader');
 var AbstractPayload = require('./abstractpayload');
+var Script = require('../../script');
 var utils = require('../../util/js');
-var BigNumber = require('bn.js');
 
 var isUnsignedInteger = utils.isUnsignedInteger;
 var isHexString = utils.isHexaString;
+var isSha256HexString = utils.isSha256HexString;
 
 var CURRENT_PAYLOAD_VERSION = 1;
 var HASH_SIZE = constants.SHA256_HASH_SIZE;
 var CKEYID_SIZE = constants.PUBKEY_ID_SIZE;
 var BLSPUBKEY_SIZE = constants.BLS_PUBLIC_KEY_SIZE;
+var BLSSIG_SIZE = constants.BLS_SIGNATURE_SIZE;
 
 /**
  * @typedef {Object} ProUpRegTransactionPayloadJSON
  * @property {number} version
- * @property {string} proTXHash
- * @property {number} mode
+ * @property {string} proTxHash
  * @property {string} pubKeyOperator
- * @property {string} keyIdVoting
- * @property {number} scriptPayoutSize
- * @property {string} scriptPayout
+ * @property {string} keyIDVoting
+ * @property {string} payoutAddress
  * @property {string} inputsHash
- * @property {number} payloadSigSize
- * @property {string} payloadSig
+ * @property {string} [payloadSig]
  */
 
 /**
  * @class ProUpRegTxPayload
  * @property {number} version uint_16	2	Upgrade Provider Transaction version number. Currently set to 1.
- * @property {string} proTXHash uint256	32	The hash of the provider transaction
+ * @property {string} proTxHash uint256	32	The hash of the provider transaction
  * @property {number} mode uint_16	2	Masternode mode
  * @property {string} pubKeyOperator BLSPubKey	48	The public key hash used for operational related signing (network messages, ProTx updates)
- * @property {string} keyIdVoting CKeyID	20	The public key hash used for voting.
+ * @property {string} keyIDVoting CKeyID	20	The public key hash used for voting.
  * @property {number} scriptPayoutSize compactSize uint	1-9	Size of the Payee Script.
  * @property {string} scriptPayout Script	Variable	Payee script (p2pkh/p2sh)
  * @property {string} inputsHash uint256	32	Hash of all the outpoints of the transaction inputs
  * @property {number} payloadSigSize compactSize uint	1-9	Size of the Signature
  * @property {string} payloadSig vector	Variable	Signature of the hash of the ProTx fields. Signed by the Owner.
  */
+/**
+ *
+ * @param {ProUpRegTransactionPayloadJSON} [payloadJSON]
+ * @constructor
+ */
 function ProUpRegTxPayload(payloadJSON) {
   AbstractPayload.call(this);
   this.version = CURRENT_PAYLOAD_VERSION;
 
   if (payloadJSON) {
-    this.proTXHash = payloadJSON.proTXHash;
-    this.mode = payloadJSON.mode;
+    this.proTxHash = payloadJSON.proTxHash;
     this.pubKeyOperator = payloadJSON.pubKeyOperator;
```

Comment:
Looking very good.
Just wanted to ask what is your reason to remove the field `mode` from the JSON representations for ProRegTx and ProUpRegTx?

Reply:
There is no `mode` field in the JSON from Dashcore's RPC. However, this field is still presented in the hex binary. So I wanted to maintain consistency with core's RPC. I think the reason why it isn't in the JSON is that there is only one mode, and this space is still allocated in the binary because it's just reserved for the future use. Can't say for sure; Maybe @codablock can clarify this

Reply:
I see, thanks Anton!